### ISSUE TITLE
Support for Snowflake Qualify clause

### ISSUE THREAD

**Issue by justinstern** [alanmcruickshank/sqlfluff#392]
It seems like even with sqlfluff configured with `dialect = snowflake` that it possibly lacks support for the Snowflake [qualify](https://docs.snowflake.com/en/sql-reference/constructs/qualify.html) clause. At least when I use the following (which structure executes correctly in Snowflake):

``` sql
1. select
2.     col1,
3.     col2
4. 
5. from some_table
6. 
7. qualify
8.     row_number() over (partition by col1 order by col1) = 1
```

linting with sqlfluff yields:

```
L:   2 | P:   5 | L026 | Reference 'some_table.col1' refers to table/view
                       | 'some_table' not found in the FROM clause or found in
                       | parent subquery.
L:   3 | P:   5 | L026 | Reference 'some_table.col2' refers to table/view
                       | 'some_table' not found in the FROM clause or found in
                       | parent subquery.
L:   7 | P:   1 | L003 | Indent expected and not found compared to line #5
L:   7 | P:   1 | L011 | Implicit aliasing of table not allowed. Use explicit
                       | `AS` clause.
L:   7 | P:   1 | L025 | Alias 'qualify' is never used in SELECT statement.
L:   7 | P:   8 |  PRS | Found unparsable section: '\n row_number() over
                       | (partition by col...'
```

This is with sqlfluff, version 0.3.4

**Comment by alanmcruickshank** [comment]
Yep, I don't think this has been implemented in the parse yet. I've never actually used the `qualify` clause, very interesting (https://docs.snowflake.com/en/sql-reference/constructs/qualify.html).

This should be very possible.

**Comment by justinstern** [comment]
@alanmcruickshank thanks for the comment. This isn't something that I've had a chance to try digging into as of yet, so if someone wants to tackle it, that'd be fine. Should I get around to it I'll drop a note here so there's no duplicate effort 😄

**Comment by aydumoulin** [comment]
I made this working locally. However I am struggling with the test suite in master so I did not feel like issuing a MR

I patched `src/sqlfluff/dialects/dialect_snowflake.py`

**Comment by alanmcruickshank** [comment]
I've started work on this one. Should have something soon.
